enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java

```diff
@@ -186,13 +184,26 @@ public void execute(@Nonnull EnforcerRuleHelper helper) throws EnforcerRuleExcep
 
       Traverser<DependencyNode> traverser = Traverser.forTree(DependencyNode::getChildren);
 
-      return StreamSupport.stream(
-              traverser.breadthFirst(resolutionResult.getDependencyGraph()).spliterator(), false)
-          .map(DependencyNode::getArtifact)
-          .filter(Objects::nonNull)
-          .map(Artifact::getFile)
-          .map(File::toPath)
-          .collect(toImmutableList());
+      DependencyNode dependencyGraph = resolutionResult.getDependencyGraph();
+      Iterable<DependencyNode> breadthFirst = traverser.breadthFirst(dependencyGraph);
+      
+      ImmutableList.Builder<Path> builder = ImmutableList.builder();
+      for (DependencyNode node : breadthFirst) {
+        // the very first one is the pom.xml where this rule appears
+        Artifact artifact = node.getArtifact();
+        if (artifact != null) { // why is this possible?
+          File file = artifact.getFile();
+          // and this very first one does not have a file; i.e. file == null
+          // but why do we care? perhaps we're assuming there is a jar file in
+          // the classpath but what we really need for this one is a classes directory
+          Path path = file.toPath();
+          builder.add(path);
+        }
+      }
+      ImmutableList<Path> built = builder.build();
+      return built;
+    } catch (NullPointerException e) {
```

Comment:
Instead of catching NPE, handle nulls in the main code. In this case, `Artifact.getFile()` can return `null` (and is so documented), so why are we throwing at all?

Reply:
good idea; done